### PR TITLE
Additional Natspec

### PR DESCRIPTION
**Pull request by bill0x2A**
Natspec for erc20 folder contracts.

Also includes minor edits for mistakes in Crowdfund.sol and Thread.sol natspec `uint256 -> uint112`.

### CODE REVIEW

File: contracts/erc20/Auction.sol [hunk 1]
```diff
@@ -61,42 +83,50 @@ contract Auction is ReentrancyGuardUpgradeable, Composable, IAuctionInitializabl
     uint64 start,
     uint32 length
   ) external override nonReentrant returns (uint256 id) {
```
Comment (bill0x2A) [inline contracts/erc20/Auction.sol:85]: This function does not actually return anything?
Also it creates multiple batches with many IDs - should this `returns (uint256 id)` just be removed?

File: contracts/dao/DAO.sol [hunk 2]
```diff
@@ -338,10 +373,15 @@ abstract contract DAO is Composable, IDAO {
...
   function _completeProposal(uint256 id, uint16 proposalType, bytes calldata data) internal virtual;
 
-  // Does not require canonically ordering when executing proposals in case a proposal has invalid actions, halting everything
-  // It would make a more robust system for specific proposal types, such as Thread's FrabricChange,
-  // if only the most recent instance of such a proposal could be edited though
-  // That is left for Thread to implement outside of the API of DAO
+  /**
+   * @notice Complete a queued proposal `id` 
+   * @param id Id of proposal to be completed
+   * @param data [Can't figuire this one out!]
```
Comment (bill0x2A) [inline contracts/dao/DAO.sol:379]: Can someone fill this in maybe? Couldn't work out the required format here.

File: contracts/dao/FrabricDAO.sol [hunk 3]
```diff
@@ -84,17 +94,32 @@ abstract contract FrabricDAO is EIP712Upgradeable, DAO, IFrabricDAO {
...
-  // a fallback function which don't error when called
-  // Between human review, function definition requirements, and the lack of privileges bestowed,
-  // this is considered to be appropriately managed
+  /**
+   * @notice Propose a contract upgrade, allows upgrading itself or any contract owned by itself
+   * @param beacon Address of beacon contract facilitating upgrades
+   * @param instance Address of contract instance to be upgraded
+   * @param version Version number
+   * @param code Address of contract containing new code
+   * @param data Data to be passed to new contract [?]
```
Comment (bill0x2A) [inline contracts/dao/FrabricDAO.sol:114]: Several parts of this contract were challenging to get right, someone more familiar should definitely check my @params are accurate here.

File: contracts/dao/FrabricDAO.sol [hunk 4]
```diff
@@ -112,26 +137,38 @@ abstract contract FrabricDAO is EIP712Upgradeable, DAO, IFrabricDAO {
...
+     * The only missing indexing case is when it's proposed to upgrade, yet that never passes/executes
+     * This should be minimally considerable and coverable by outside solutions if truly needed
+     */
     emit UpgradeProposal(id, beacon, instance, version, code, data);
   }
 
+  /**
+   * @notice Create a proposal to mint, transfer, cancel a sell order, auction tokens, or any combination of these
+   * @param token Address of token to be proposed on
+   * @param target Target address for proposed action output
```
Comment (bill0x2A) [inline contracts/dao/FrabricDAO.sol:166]: Not 100% on these params

File: contracts/dao/FrabricDAO.sol [hunk 5]
```diff
@@ -112,26 +137,38 @@ abstract contract FrabricDAO is EIP712Upgradeable, DAO, IFrabricDAO {
...
+   * @param price Price of tokens for proposed action
+   * @param amount Quantity of tokens to perform action on
+   * @param info Any extra information
```
Comment (bill0x2A) [inline contracts/dao/FrabricDAO.sol:169]: Not too confident on these

**Review by zeryx — APPROVED** [review]

**Pull request merged**